artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/WildcardAddressManager.java

```diff
@@ -175,36 +175,38 @@ public void clear() {
       wildCardAddresses.clear();
    }
 
-   private Address addAndUpdateAddressMap(final SimpleString address, boolean getBiased) {
+   private Address addAndUpdateAddressMap(final SimpleString address) {
       final boolean containsWildCard = address.containsEitherOf(wildcardConfiguration.getAnyWords(), wildcardConfiguration.getSingleWord());
       final Map<SimpleString, Address> addressMap = containsWildCard ? wildCardAddresses : addresses;
-      Address actualAddress = null;
-      if (getBiased) {
-         // CHM::get doesn't need to synchronize anything, so it's to be preferred for getBiased cases
-         actualAddress = addressMap.get(address);
-      }
+      Address actualAddress = addressMap.get(address);
       if (actualAddress == null) {
-         actualAddress = addressMap.computeIfAbsent(address, addressKey -> new AddressImpl(addressKey, wildcardConfiguration));
-      }
-      assert actualAddress.containsWildCard() == containsWildCard;
-      synchronized (this) {
-         if (containsWildCard) {
-            for (Address destAdd : this.addresses.values()) {
-               if (destAdd.matches(actualAddress)) {
-                  destAdd.addLinkedAddress(actualAddress);
-                  actualAddress.addLinkedAddress(destAdd);
-               }
-            }
-         } else {
-            for (Address destAdd : wildCardAddresses.values()) {
-               if (actualAddress.matches(destAdd)) {
-                  destAdd.addLinkedAddress(actualAddress);
-                  actualAddress.addLinkedAddress(destAdd);
+         synchronized (this) {
+            actualAddress = addressMap.get(address);
```

Comment:
Same here: https://github.com/apache/activemq-artemis/pull/3352/files#r527734942

Reply:
that is the crux, one fast return path is good. 2 now gives us in total 3 return paths. I think it is more readable with a single return path.